android/src/main/java/com/mediamonks/rnaccengage/RNAccengageModule.java

```diff
@@ -52,6 +64,11 @@ public void trackEvent(int key) {
         A4S.get(getReactApplicationContext()).trackEvent(key);
     }
 
+    @ReactMethod
+    public void getDeviceID(Callback callback) {
+        callback.invoke(accengageId);
```

Comment:
What happens if accengageId is still null?
Whuy is the id not requested here, it is an async call already.

Reply:
Thanks for pointing this out. A null value wouldn't translate to a null ID for the callback, because the parameter for invoke() is an array with the actual JS parameters. A null value wouldn't even mean anything in JS land, and it'd be a sure crash.